packages/gooddata-eval/src/gooddata_eval/core/agentic/metric_skill.py

```
@@ -127,6 +127,23 @@ def _extract_metric_result(tool_call_events: list[ToolCallEvent]) -> dict | None
+
+    Eval runs share a persistent workspace, so a metric left behind is picked up by
+    a later test — the agent reuses it (returning ``SELECT {id}`` instead of full
+    MAQL) and the assertion fails. Deleting the created metric on the way out keeps
+    the workspace clean for the next run. Mirrors ``alert_skill._delete_alert``.
+    """
+    host = str(client._base).split("/api/")[0]
+    url = f"{host}/api/v1/entities/workspaces/{workspace_id}/metrics/{metric_id}"
+    try:
+        response = client._client.delete(url, headers=client._auth)
+        response.raise_for_status()
+    except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)